homeworks/taridaniel/9_fastapi/fastapi_homework_api.py

```diff
@@ -0,0 +1,75 @@
+from fastapi import FastAPI, HTTPException, Depends
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.future import select
+from sqlalchemy.exc import NoResultFound
+from models import HouseholdAppliances, HouseholdAppliancesResponse, HouseholdAppliancesResponse
+from database import get_db, engine, Base
+import uuid
+from typing import List, Optional
+
+app = FastAPI()
+
+# Create the database tables
+@app.on_event("startup")
+async def startup():
+    async with engine.begin() as conn:
+        await conn.run_sync(Base.metadata.create_all)
+
+class HouseholdAppliancesRequest:
+    item_name: str
+    quantity: int
+    price: float
+    category: Optional[str] = None
+
+class HouseholdAppliancesResponse(HouseholdAppliancesRequest):
+    id: str
+
+# List all products
+@app.get("/products/", response_model=List[HouseholdAppliancesResponse])
+async def read_products(skip: int = 0, limit: int = 10, db: AsyncSession = Depends(get_db)):
+    result = await db.execute(select(HouseholdAppliances).offset(skip).limit(limit))
+    products = result.scalars().all()
+    return products
+
+# List one product by id
+@app.get("/products/{product_id}", response_model=HouseholdAppliancesResponse)
+async def read_product(product_id: str, db: AsyncSession = Depends(get_db)):
+    result = await db.execute(select(HouseholdAppliances).filter(HouseholdAppliances.id == product_id))
+    product = result.scalars().first()
+    if not product:
+        raise HTTPException(status_code=404, detail="Product not found")
+    return product
+
+# Add one product
+@app.post("/products/", response_model=HouseholdAppliancesResponse)
```

Comment:
figyelj hogy ebben a file-ban több helyen is `HouseholdAppliancesRequest`-et használsz, de a models.py-ban nem így hívod, ott nincs s betű az Appliance után. Ezeket egységesítsd.